### PR TITLE
autoconf: fix package info on windows

### PR DESCRIPTION
**Pull request by SpaceIm**
Fix regression on windows introduced in https://github.com/conan-io/conan-center-index/pull/12896

We must still use legacy unix_path in package_info().
see: https://github.com/conan-io/conan/issues/12499 & https://github.com/conan-io/conan/issues/12192#issuecomment-1308589581

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/autoconf/all/conanfile.py [hunk 1]
```diff
@@ -68,64 +73,66 @@ def generate(self):
...
+        self._patch_sources()
         autotools = Autotools(self)
         autotools.configure()
         autotools.make()
 
     def package(self):
         autotools = Autotools(self)
-        autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])  # Need to specify the `DESTDIR` as a Unix path, aware of the subsystem
+        # TODO: can be replaced by autotools.install() if required_conan_version = ">=1.54.0"
+        autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
```
Comment (jellespijker) [inline recipes/autoconf/all/conanfile.py:93]: Maybe add a FIXME comment once CCI uses the Conan version with your pr https://github.com/conan-io/conan/pull/12193

File: recipes/autoconf/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,53 +1,51 @@
...
+import os
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    exports_sources = "configure.ac", "config.h.in", "Makefile.in", "test_package_c.c", "test_package_cpp.cpp",
-    generators = "AutotoolsDeps", "AutotoolsToolchain", "VirtualBuildEnv"
+    settings = "os", "arch", "compiler", "build_type"
     test_type = "explicit"
+    win_bash = True
```
Comment (jellespijker) [inline recipes/autoconf/all/test_package/conanfile.py:14]: Maybe move this to the build_requirements method for consistency

File: recipes/autoconf/all/test_v1_package/conanfile.py [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
-import shutil
-
-required_conan_version = ">=1.45.0"
 
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    exports_sources = "../test_package/configure.ac", "../test_package/config.h.in", "../test_package/Makefile.in", "../test_package/test_package_c.c", "../test_package/test_package_cpp.cpp",
+    settings = "os", "arch", "compiler", "build_type"
     test_type = "explicit"
```
Comment (jellespijker) [inline recipes/autoconf/all/test_v1_package/conanfile.py:11]: Shouldn't the `win_bash` be set as well for `test_v1_package` ?

**Review by jellespijker — APPROVED** [review]
Couple of minor remarks questions, but overall it looks good

File: recipes/autoconf/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/autoconf/all/conanfile.py:93]: it's fixed in 1.54.0 so for me the TODO comment has already everything

File: recipes/autoconf/all/test_package/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/autoconf/all/test_package/conanfile.py:14]: no, it doesn't work in test package

File: recipes/autoconf/all/test_v1_package/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/autoconf/all/test_v1_package/conanfile.py:11]: no, in conan v1 it's a different mechanism with win_bash in self.run directly.

**Pull request merged**